### PR TITLE
Extension/views

### PR DESCRIPTION
**Pull request by rishabh-vinsol**

### CODE REVIEW

File: lib/tasks/port_legacy_products.rake [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace :port_legacy_products do
+    task :set_category_default => :environment do
```
Comment (himanshumishra31) [inline lib/tasks/port_legacy_products.rake]: fix identation

File: db/migrate/20230414065112_create_address.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateAddress < ActiveRecord::Migration[7.0]
+  def change
+    create_table :addresses do |t|
+      t.string :state
+      t.string :city
+      t.string :country
+      t.string :pincode
+      t.references :user, null: false, foreign_key: true
```
Comment (himanshumishra31) [inline db/migrate/20230414065112_create_address.rb:8]: check what all constraints references add

File: app/models/product.rb [hunk 3]
```diff
@@ -21,6 +22,8 @@ class Product < ApplicationRecord
   validates :image_url, allow_blank: true, url: true
   validates :permalink, uniqueness: true, format: { with: PERMALINK_FORMAT }
   validates_length_of :words_in_permalink, minimum: 3
+  validates :category_id, presence: true
```
Comment (himanshumishra31) [inline app/models/product.rb]: This is not required as already handled by belongs_to

File: app/models/user.rb [hunk 4]
```diff
@@ -8,6 +8,8 @@ def validate_each(record, attribute, value)
 class User < ApplicationRecord
   has_many :orders, dependent: :destroy
```
Comment (himanshumishra31) [inline app/models/user.rb:13]: beautify model file

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,4 +1,5 @@
 class UsersController < ApplicationController
+  layout 'my_orders', only: [:orders, :line_items]
```
Comment (himanshumishra31) [inline app/controllers/users_controller.rb:2]: use symbol

**Review by himanshumishra31 — COMMENTED** [review]
comment added

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (rishabh-vinsol) [reply app/controllers/users_controller.rb:2]: If we want to define a specific layout we have to pass string only, if we pass symbol, it searches for any variable or function not the layout

File: app/models/product.rb (hunk 3 above)
Comment (rishabh-vinsol) [reply app/models/product.rb]: Done

File: db/migrate/20230414065112_create_address.rb (hunk 2 above)
Comment (rishabh-vinsol) [reply db/migrate/20230414065112_create_address.rb:8]: null option defaults to true for references

File: app/models/user.rb (hunk 4 above)
Comment (rishabh-vinsol) [reply app/models/user.rb:13]: Done

**Pull request merged**